src/js/menu/menu-item.js

```diff
@@ -74,6 +74,15 @@ class MenuItem extends ClickableComponent {
       textContent: this.localize(this.options_.label)
     });
 
+    const containsHexCode = (s) => {
```

Comment:
This feels like a utility function that pertains to strings in particular so I'm wondering if it might be better to move it [here](https://github.com/videojs/video.js/blob/main/src/js/utils/str.js).

Since we're proposing using innerHTML as part of this solution, and this function is essentially guarding that step, maybe it would be beneficial to add some tighter conditions to make it more difficult to use other problematic characters (i.e.: `/'"`, etc) that are typically escaped for safety? We should also probably add a check for a semicolon `;` as part of the regex test.